py-glaredb/src/logical_plan.rs

```diff
@@ -60,6 +60,9 @@ impl PyLogicalPlan {
     fn to_pandas(&self, py: Python) -> PyResult<PyObject> {
         self.collect(py)?.to_pandas(py)
     }
+    fn show(&self, py: Python, show_metadata: bool, n_cols: u64) -> PyResult<()> {
+        self.collect(py)?.show(py, show_metadata, n_cols)
```

Comment:
for the logical_plan, we should push down the `n_rows` to the plan. No sense in over fetching data 😃